src/python/pants/backend/python/target_types.py

```diff
@@ -1349,3 +1350,77 @@ class PythonDistribution(Target):
         See {doc_url('python-distributions')}.
         """
     )
+
+
+class SetuptoolsSCMDummySourceField(OptionalSingleSourceField):
+    """A dummy SourceField for participation in the codegen machinery."""
+
+    alias = "_dummy_source"  # Leading underscore omits the field from help.
+    help = "A version generated by setuptools_scm"
+
+
+class SetuptoolsSCMTagRegexField(StringField):
+    default = None
+    alias = "tag_regex"
+    help = softwrap(
+        """
+        A Python regex string to extract the version part from an SCM tag.
+
+        The regex needs to contain either a single match group, or a group named version,
+        that captures the actual version information.
+
+        Note that this is unrelated to the tags field and to Pants's own tags concept.
+
+        See https://github.com/pypa/setuptools_scm for details.
+        """
+    )
+
+
+class SetuptoolsSCMWriteToField(StringField):
+    required = True
+    alias = "write_to"
+    help = "Generate the version data to this relative path."
+
+
+class SetuptoolsSCMWriteToTemplateField(StringField):
+    """A format string that is given the current version as the `version` format kwarg.
+
+    E.g., 'version = "{version}"'
+
+    See See https://github.com/pypa/setuptools_scm
+    """
+
+    required = True
+    alias = "write_to_template"
```

Comment:
Ditto naming, although less clear what to call this.

Also: it seems reasonable for this to maybe have a name?

Reply:
The original names were just copied from setuptools_scm's own args, but I agree we can use different names, since we're not actually using that part of their functionality.